### ISSUE TITLE
💡 Ideas

### ISSUE THREAD

**Issue by alfg** [alfg/opendrinks#50]
Looking for ideas, suggestions and feedback for https://opendrinks.netlify.com. 

Feel free to post them here.

**Comment by ArelySkywalker** [comment]
Some ideas I have (I have a similar app that i built!) 😄:

- Recommended Serving Glass
- Search by Ingredient
- Type of drink (Cocktail, Juice, Coffee, Tea, etc)

I would love to help contribute for Hacktober Fest! 🎃

**Comment by alfg** [comment]
Good ideas @ArelySkywalker!

I plan on opening a few feature requests for search filtering and adding tags to the drink recipe json. Recommended Serving Glass would be nice as an optional field.

**Comment by alfg** [comment]
#101 and #102 created. Feel free to take on any of these if you like!

**Comment by ArelySkywalker** [comment]
Awesome, thanks!!

**Comment by SorielV** [comment]
Some tools that may be useful.
[Gridsome](https://gridsome.org]) JamStack
[Vuex](https://vuex.vuejs.org/) Store

**Comment by alfg** [comment]
Hi @SorielV,

I am not too familiar with Gridsome, but checking into it.

For Vuex, I'm not sure it seems necessary to use a centralized store for this app yet, but where do you feel it might be needed?

**Comment by TobyAsE** [comment]
Add the capability to export a chosen recipe. Either as a printable page/pdf or machine-readable (just dump the json...)

**Comment by VVNoodle** [comment]
Add a similar drinks section for each recipe. Similar drinks are drinks that have common keywords/ingredients. We can limit the number of recommendation and pick the ones that have the most similarities.

**Comment by alfg** [comment]
Good ideas @TobyAsE and @VVNoodle. I've added #133 and #134 respectively.

**Comment by TobyAsE** [comment]
A list view where all the drinks are shown. Maybe in list form or as a tiled display.
Should be sortable by keywords.

**Comment by alfg** [comment]
@TobyAsE This is currently the explore page, but hopefully we can get pagination added first, then we can add filters/sorting. I added #163 as the feature request.

**Comment by TobyAsE** [comment]
Wow, @stevenanthonyrevo was fast.
Once #175 is merged, we can add the keyword filter.

**Comment by TobyAsE** [comment]
As @VVNoodle said in #173 , I also think that it would be a great idea to seperate the ingredients into multiple parts. `name` and `quantity` are obvious, I would also add something like `comment`.
This could be quite a task to apply to all recipes now, but it's just a bit of tedious work.
This would make searching and comparing way easier down the road.

**Comment by alfg** [comment]
Yeah, I like the idea of updating ingredients into `name` and `quantity`. But yes, it can be a tedious task to update the existing ones. Or a one-off script to update all of them.

**Comment by Assisting** [comment]
I could take a crack at writing something for that, I've done some similar work in the past in Python.

**Comment by alfg** [comment]
> 
> 
> I could take a crack at writing something for that, I've done some similar work in the past in Python.
That would be great! Feel free to try and let us know!

**Comment by Assisting** [comment]
Well, the results are in PR #186. I feel like it went OK, although some of the weights and measures were pretty loose in these ingredient lists and so I didn't try to normalize the inputs or anything.

**Comment by alfg** [comment]
Thanks @Assisting! I've updated the [CONTRIBUTING](https://github.com/alfg/opendrinks/blob/master/CONTRIBUTING.md) docs with the updated JSON schema for ingredients!

**Comment by TobyAsE** [comment]
I just added prettier to my project locally and I am getting screamed at. Maybe we should require a linter in the project to keep the style consistent across the codebase.

**Comment by alfg** [comment]
Hey @TobyAsE, would this be different from the supplied `.eslintrc` file? I think this was part of the `vue-cli` generator. 

I've been re-linting some of the warnings popping up occasionally.

**Comment by TobyAsE** [comment]
Ah, I see. Somehow, eslint didn't recognize that (or I am stupid). It's working now an I'm occasionally getting some warnings. Could you create an issue for the re-linting, we could move this conversation there? ;)

**Comment by alfg** [comment]
Sure, @TobyAsE. Created #225

**Comment by kcusaac** [comment]
What about adding a feature for favoriting drinks? It can be stored in the browser.

**Comment by TobyAsE** [comment]
@kcusaac sounds like a good idea to me.
If the user has stored some drinks as favourite, we can display them quick and easy on the main page.

**Comment by shultztom** [comment]
Would it make sense to have a .nvmrc or specify the version of node in the package.json? With this many contributors, it might be nice to all be on the same version of node.js

**Comment by alfg** [comment]
@kcusaac This is a great idea and something I had in mind to propose as a feature request. Something like saving favorites to local storage by "starring" a drink. I'll add the feature request today.

**Comment by alfg** [comment]
@shultztom Yes, good call. I'll add one with the latest LTS version (v10.16.3)

**Comment by shultztom** [comment]
> @shultztom Yes, good call. I'll add one with the latest LTS version (v10.16.3)

I've made a PR here if would want to look at that

https://github.com/alfg/opendrinks/pull/232

**Comment by alfg** [comment]
@shultztom Thanks, checking it out now.

**Comment by alfg** [comment]
Another idea as the project continues to grow in features would be to add unit testing. Jest and Mocha seem to be supported for component and unit testing.

**Comment by VVNoodle** [comment]
I'd be down to initiate a test environment. Should we create a __tests__ folder mimicking the src folder, or have the test side-by-side with the file to be tested?

**Comment by TobyAsE** [comment]
To make contributing a recipe easier we could build a recipe-builder with a nice ui that generates the json of a recipe.

**Comment by alfg** [comment]
> 
> 
> I'd be down to initiate a test environment. Should we create a **tests** folder mimicking the src folder, or have the test side-by-side with the file to be tested?

I'd say in a `tests` folder.

**Comment by alfg** [comment]
> 
> 
> To make contributing a recipe easier we could build a recipe-builder with a nice ui that generates the json of a recipe.

This is cool idea, but I feel if we go that route, we might as well build out a full API backend (which is out of scope for this project) since that's kinda creating a form for it, ha. However, I'm trying to keep the project strictly a static website for now.

For Hacktoberfest, I like the idea of recipes being submitted as JSON by users, but we can revisit this in the future. 

Maybe some kind of recipe schema validation tool? Or even some kind of bot that validates the PR. 💡

**Comment by TobyAsE** [comment]
I'll have a look into creating a validator.
A bot sounds like a nice idea

**Comment by VVNoodle** [comment]
> I'd say in a tests folder.

I've create an issue for creating the test environment. Can you assign it to me? Thanks!

**Comment by alfg** [comment]
@VVNoodle looks like there's a PR up for it on #265 as a starting point. Is this similar to what you had in mind too?

**Comment by VVNoodle** [comment]
Yeah my bad didn't see that. I thought PRs should correspond to existing Issues.

**Comment by Chrispy404** [comment]
> Yeah my bad didn't see that. I thought PRs should correspond to existing Issues.

My bad @VVNoodle I got a little ahead of myself. I saw mention of testing in the ideas thread here the previous day, but there was never an issue made so i figured it had been overlooked or forgotten about and quickly put something together. As @alfg said, what I've put up is simply a starting point so if you're wanting to work on tests there's still plenty to do.

**Comment by TobyAsE** [comment]
We're getting a few warnings when `npm install`, should we do something about this?

**Comment by TobyAsE** [comment]
There is nothing we do with @kcusaac s favoring-Feature right now. We should change that. Output the favorites somewhere.

**Comment by alfg** [comment]
@TobyAsE I believe @kcusaac mentioned he would create the Favorites view once #239 was merged. @kcusaac are you working on this?

**Comment by alfg** [comment]
> 
> 
> We're getting a few warnings when `npm install`, should we do something about this?


@TobyAsE I updated `package.json` to clear up 2 of the warnings from missing fields. Not sure how to fix the other 2, but I think they are minor and shouldn't really affect anything.

**Comment by kcusaac** [comment]
> @TobyAsE I believe @kcusaac mentioned he would create the Favorites view once #239 was merged. @kcusaac are you working on this?

Yeeh, I'll push it tomorrow. Do you think we should have a new tab with the favorites?

**Comment by alfg** [comment]
@kcusaac Sure. A new tab is fine.

**Comment by alfg** [comment]
Another idea and something I've been working on is to create a mobile app using the same recipe data as a "recipes feed" for the mobile app to fetch its data from. 

I'm currently experimenting with React-Native/Expo to create the mobile app, and using Netlify's "Functions" service to implement "feeds" for the mobile app to consume.

Also, Netlify has also upgraded this project to their Open Source plan to help sponsor the project! 🎉

**Comment by alfg** [comment]
Another idea is eventually I would like to replace the Home/Random view with something like a "Featured" view. In this view we can display a curated list of drinks configured by a JSON schema.

It would be cool to create customized lists of drinks, by popularity, themed, seasonal, etc... Like Halloween based drinks. :) It would give it more of a true "homepage" feel.

**Comment by alfg** [comment]
https://opendrinks.io is registered and live!

**Comment by alfg** [comment]
Thanks for all the ideas! This was a fun project!

**Comment by Assisting** [comment]
Good job team! Thanks for having us along for the ride @alfg! :clap:

**Comment by VVNoodle** [comment]
Thanks @alfg for maintaining the project this past month and every contributors here for keeping the project active. Definitely learned quite a lot on contributing open source projects!
